### PR TITLE
feat: rework webhook endpoints

### PR DESCRIPTION
**Pull request by JoeCap08055**
- Refactor controllers into controllers/v1 directory structure
- Split out webhook endpoint from graph endpoints
- Add endpoints to get/remove webhooks by specific MSA or URL
- Rework caching of webhooks to take advantage of Redis hashes and JS Set to avoid duplication of webhook registrations
- Refactor e2e tests for new API

e2e tests now pass

### CODE REVIEW

File: apps/api/src/controllers/v1/webhooks-v1.controller.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  @Get('urls')
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'Get all webhooks registered to the specified URL' })
+  @ApiOkResponse({ description: 'Retrieved all webhooks registered to the specified URL' })
+  async getWebhooksForUrl(@Query('url') url: string): Promise<string[]> {
+    return this.apiService.getWatchedGraphsForUrl(url);
+  }
+
+  @Put()
+  @ApiOperation({ summary: 'Watch graphs for specified dsnpIds and receive updates' })
```
Comment (wilwade) [inline apps/api/src/controllers/v1/webhooks-v1.controller.ts:43]: @JoeCap08055 We were talking a bit about potential missing endpoints. I think a potential missing endpoint here is one to register a webhook for ALL users.

So specific to the renaming, if we were to create such an endpoint, would we modify this endpoint structure in anyway? (Or just change the Dto to have that array optional to mean "all"?)

Comment (JoeCap08055) [reply apps/api/src/controllers/v1/webhooks-v1.controller.ts:43]: I was thinking the same thing; I think we can just add an interpretation to the DTO for the existing endpoint, such that a zero MSA ID means "register for all users". Or if that seems too hacky, we could modify the DTO in some other way, but make it additive/backward-compatible. No need for a separate, dedicated endpint IMO.

File: apps/api/src/api.service.ts [hunk 2]
```diff
@@ -27,7 +49,7 @@ export class ApiService implements BeforeApplicationShutdown {
 
   beforeApplicationShutdown(_signal?: string | undefined) {
     try {
-      this.redis.del(QueueConstants.REDIS_WATCHER_PREFIX);
+      // this.redis.del(QueueConstants.REDIS_WATCHER_PREFIX);
```
Comment (JoeCap08055) [inline apps/api/src/api.service.ts]:
```suggestion
```

**Review by mattheworris — APPROVED** [review]
Pair reviewed.
🚢 it!

**Pull request merged**